net.sourceforge.tomatoes.IHaveNoTomatoes.json

```
@@ -0,0 +1,111 @@
+{
+  "app-id": "net.sourceforge.tomatoes.IHaveNoTomatoes",
+  "default-branch": "stable",
+  "runtime": "org.freedesktop.Platform",
+  "runtime-version": "22.08",
+  "sdk": "org.freedesktop.Sdk",
+  "command": "tomatoes",
+  "cleanup": [
+    "/include",
+    "*.a",
+    "*.la",
+    "/bin/libmikmod-config",
+    "/bin/sdl-config",
+    "/lib/pkgconfig",
+    "/share/aclocal",
+    "/share/info",
+    "/share/man"
+  ],
+  "finish-args": [
+    "--socket=wayland",
+    "--socket=fallback-x11",
+    "--socket=pulseaudio",
+    "--device=dri",
+    "--persist=.tomatoes"
+  ],
+  "modules": [
+    "shared-modules/glu/glu-9.json",
+    "shared-modules/SDL/sdl12-compat.json",
+    "shared-modules/SDL/SDL_image-1.2.12.json",
+    {
+      "name": "libmikmod-2",
+      "buildsystem": "autotools",
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://sourceforge.net/projects/mikmod/files/libmikmod/3.1.21.1/libmikmod-3.1.21.1.tar.gz",
+          "sha256": "1a88922b5a67c308899c1d991ddfb62b5d851f8a94175612b33856854938b2fb",
+          "x-checker-data": {
+              "type": "html",
+              "url": "https://sourceforge.net/projects/mikmod/rss",
+              "pattern": "<link>(https://sourceforge.net/.+/libmikmod-3\.1\.([\\d\\.]+\\d).tar.gz)/download</link>"
+          }
+        }
+      ]
+    },
+    "shared-modules/SDL/SDL_mixer-1.2.12.json",
+    {
+      "name": "tomatoes-data",
+      "buildsystem": "simple",
+      "build-commands": [
+        "mkdir -p /app/share/tomatoes/",
+        "cp -r config.cfg music tomatoes.mpk /app/share/tomatoes/"
+      ],
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://downloads.sourceforge.net/tomatoes/tomatoes-linux-1.5.tar.bz2",
+          "sha256": "69a3af1994cb58409be9b469e1aa4bdb91c405d99071c7431526640ec53d1300"
+        }
+      ]
+    },
+    {
+      "name": "tomatoes",
+      "buildsystem": "simple",
+      "build-commands": [
+        "make",
+        "cp tomatoes /app/bin/",
+        "mkdir -p /app/share/metainfo",
+        "mkdir -p /app/share/applications",
+        "mkdir -p /app/share/icons/hicolor/64x64/apps",
+        "cp net.sourceforge.tomatoes.IHaveNoTomatoes.appdata.xml /app/share/metainfo/",
+        "cp net.sourceforge.tomatoes.IHaveNoTomatoes.desktop /app/share/applications/",
+        "cp icon.png /app/share/icons/hicolor/64x64/apps/net.sourceforge.tomatoes.IHaveNoTomatoes.png"
```

```suggestion
        "install -Dm644 net.sourceforge.tomatoes.IHaveNoTomatoes.appdata.xml -t /app/share/metainfo/",
        "install -Dm644 net.sourceforge.tomatoes.IHaveNoTomatoes.desktop -t /app/share/applications/",
        "install -Dm644 icon.png /app/share/icons/hicolor/64x64 -t /apps/net.sourceforge.tomatoes.IHaveNoTomatoes.png"
```

also the minimum size required is 128x128.

---

> also the minimum size required is 128x128.

As far as I know, no larger size exists.

---

gotta have to either
1. upscale the icon to 128x128
OR
2. add a `flathub.json` like this https://github.com/flathub/org.freedesktop.LinuxAudio.Plugins.OBXd/blob/branch/21.08/flathub.json

---

I assume there is no SVG either.

---

> gotta have to either

Which would you consider more appropriate? I know some places are very particular about not misrepresenting what you have to the automated checkers.

> I assume there is no SVG either.

I seriously doubt it. Aside from the minor patches by distro maintainers to account for things that shouldn't have been done in the first place (hard-coding `-march=pentium` in the makefile, using `./` for asset paths, etc.), the game was last updated in 2004 and the icon looks like it probably was rendered directly to a raster image from something like 3D Studio Max.

---

2. is fine given the announced status of the project. it will cause the validation to just ignore.